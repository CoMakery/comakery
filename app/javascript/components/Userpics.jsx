import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`

const Users = styled.div`
  margin-right: 8px;
`

const Userpic = styled.img`
  border-radius: 50%
  width: 15px;
  height: 15px;
  border: solid 1px #ffffff;
  margin-right: -5px;
`

const More = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
`

class Userpics extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Wrapper>
          <Users>
            {this.props.pics.slice(0, this.props.limit).map((u, i) =>
              <Userpic key={i} src={u} />
            )}
          </Users>

          {this.props.pics.length - this.props.limit > 0 &&
          <More>
            {`+${this.props.pics.length - this.props.limit}`}
          </More>
          }
        </Wrapper>
      </React.Fragment>
    )
  }
}

Userpics.propTypes = {
  pics : PropTypes.array,
  limit: PropTypes.number
}
Userpics.defaultProps = {
  pics : [],
  limit: 3
}
export default Userpics
