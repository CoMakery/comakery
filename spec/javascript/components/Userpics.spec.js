import React from 'react'
import { mount } from 'enzyme'
import Userpics from 'components/Userpics'

describe('Userpics', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<Userpics />)

    expect(wrapper).toMatchSnapshot()
  })
})
