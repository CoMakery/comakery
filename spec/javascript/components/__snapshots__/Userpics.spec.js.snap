// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Userpics renders component correctly 1`] = `
<Userpics
  limit={3}
  pics={Array []}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "jixqaJ",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
",
            ],
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa jixqaJ"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "mNJvu",
                  "rules": Array [
                    "
  margin-right: 8px;
  display: flex;
",
                  ],
                },
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH mNJvu"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Userpics>
`;
